### ISSUE TITLE
twisted.web.http.Request.parseCookies tests pass None to a flag, should pass False instead

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6734]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)| @lvh reported|
|-|-|
|Trac ID|trac#6734|
|Type|defect|
|Created|2013-09-09 14:16:44Z|
|Branch|https://github.com/twisted/twisted/tree/flags-6734|

From [#6119](https://github.com/twisted/twisted/issues/6119)#comment:11:

The branch is also introducing tests that pass None as a false value for a flag parameter (Request.__init__ queued). This works but is weird. The code should pass False instead. 

<details><summary>Searchable metadata</summary>

```
trac-id__6734 6734
type__defect defect
reporter__lvh lvh
priority__normal normal
milestone__ 
branch__branches_flags_6734 branches/flags-6734
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__easy easy
time__1378736204000000 1378736204000000
changetime__1378749200000000 1378749200000000
version__None None
owner__lvh lvh
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)|@lvh set owner to @lvh<br>@lvh set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_4"></a>|@lvh commented|
|-|-|

(In [39899]) Branching to 'flags-6734'

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)|@lvh removed owner<br>@lvh set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_6"></a>|@lvh commented|
|-|-|

Buildbots: http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/flags-6734

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_7"></a>|@wallrj set owner to @lvh|
|-|-|

Thanks lvh,

The changes look good.

I noticed a few remaining cases of queued=None

```
[richard@zorin flags-6734]$ ack 'http\.Request\(.+None\)'
twisted/web/test/test_http.py
295:    request = http.Request(DummyChannel(), None)
1164:        req = http.Request(DummyChannel(), None)
1196:        req = http.Request(DummyChannel(), None)
1206:        req = http.Request(DummyChannel(), None)
1216:        req = http.Request(DummyChannel(), None)
1225:        req = http.Request(DummyChannel(), None)
1235:        req = http.Request(DummyChannel(), None)
1244:        req = http.Request(DummyChannel(), None)
1255:        req = http.Request(channel, None)
1269:        req = http.Request(channel, None)
1282:        req = http.Request(DummyChannel(), None)
1292:        req = http.Request(DummyChannel(), None)
1304:        req = http.Request(DummyChannel(), None)
1317:        req = http.Request(d, None)
1328:        req = http.Request(DummyChannel(), None)
1341:        req = http.Request(d, None)
1351:        req = http.Request(DummyChannel(), None)
1361:        req = http.Request(DummyChannel(), None)
1383:        req = http.Request(DummyChannel(), None)
1415:        req = http.Request(DummyChannel(), None)
1440:        req = http.Request(DummyChannel(), None)
```

It seems like a good opportunity to fix these tests too - otherwise someone will come along and copy the bad tests again.

Please merge after fixing those remaining cases and linking to some clean build results.

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_8"></a>|@lvh set status to `closed`|
|-|-|

(In [39916]) Merge flags-6734: use False instead of None as a flag value in test_http

Author: lvh
Reviewer: rwall
Fixes: [#6734](https://github.com/twisted/twisted/issues/6734)

test_http used None as a value for Request's queued argument, which is a flag. It uses False now.
